### PR TITLE
Extend IAM fetcher to include password policy

### PR DESCRIPTION
**Pull request by uri-weisman**
**What does this PR do?**
Extend the IAM fetcher to include the password policy resource.
A required resource to be evaluated by rules: 1_8 and 1_9 of the CIS AWS benchmark.

**Related Issues**
Can be tested with:
- https://github.com/elastic/csp-security-policies/pull/150

**Checklist**
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/providers/awslib/config_provider.go [hunk 1]
```diff
@@ -21,26 +21,29 @@ import (
...
+func (p ConfigProvider) InitializeAWSConfig(ctx context.Context, cfg aws.ConfigAWS, log *logp.Logger) (awssdk.Config, error) {
 	awsConfig, err := aws.InitializeAWSConfig(cfg)
 	if err != nil {
 		return awssdk.Config{}, err
 	}
+
 	metadata, err := p.MetadataProvider.GetMetadata(ctx, awsConfig)
 	if err != nil {
-		return awssdk.Config{}, err
+		log.Errorf("MetadataProvider.GetMetadata Error: %v, setting AWSConfig region to default - %s", err, DefaultRegion)
```
Comment (uri-weisman) [inline resources/providers/awslib/config_provider.go:42]: Not sure I'm happy with this logic.
If we failed to query the machine metadata we'll fallback to the default region - "us-east-1".
It was mainly changed for development, running cloudbeat as a standalone process.

File: go.mod [hunk 2]
```diff
@@ -264,9 +263,11 @@ replace (
 	github.com/apoydence/eachers => github.com/poy/eachers v0.0.0-20181020210610-23942921fe77 //indirect, see https://github.com/elastic/beats/pull/29780 for details.
 	github.com/dop251/goja => github.com/andrewkroh/goja v0.0.0-20190128172624-dd2ac4456e20
 	github.com/dop251/goja_nodejs => github.com/dop251/goja_nodejs v0.0.0-20171011081505-adff31b136e6
+	github.com/elastic/csp-security-policies => /Users/uriweisman/development/csp/csp-security-policies
```
Comment (uri-weisman) [inline go.mod:266]: Will be removed before the merge

File: deploy/aws/cloudbeat-aws.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  runtime_cfg:
+    activated_rules:
+      cis_aws:
+        - cis_1_8
+        - cis_1_9
+# =================================== Kibana ===================================
+setup.kibana:
+
+  # Kibana Host
+  host: "http://localhost:5601"
```
Comment (oren-zohar) [inline deploy/aws/cloudbeat-aws.yml:20]:
```suggestion
  host: "http://host.docker.internal:5601"
```

File: deploy/aws/cloudbeat-aws.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+#cloud.id:
+
+# The cloud.auth setting overwrites the `output.elasticsearch.username` and
+# `output.elasticsearch.password` settings. The format is `<user>:<pass>`.
+#cloud.auth:
+
+# ---------------------------- Elasticsearch Output ----------------------------
+output.elasticsearch:
+  # Array of hosts to connect to.
+  hosts: ["localhost:9200"]
```
Comment (oren-zohar) [inline deploy/aws/cloudbeat-aws.yml:37]:
```suggestion
  hosts: ${ES_HOST}
```

File: deploy/aws/cloudbeat-aws.yml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+  # Protocol - either `http` (default) or `https`.
+#  protocol: "https"
+
+  # Authentication credentials - either API key or username/password.
+  #api_key: "id:api_key"
+#  username: ${ES_USERNAME}
+#  password: ${ES_PASSWORD}
```
Comment (oren-zohar) [inline deploy/aws/cloudbeat-aws.yml:45]: uncomment?

File: deploy/aws/cloudbeat-aws.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  # Enable to allow sending output to older ES versions
+  allow_older_versions: true
+
+# ================================= Processors =================================
+processors:
+  - add_host_metadata: ~
+  - add_cloud_metadata: ~
+  - add_docker_metadata: ~
+
```
Comment (oren-zohar) [inline deploy/aws/cloudbeat-aws.yml:55]:
```suggestion
  - add_host_metadata: ~
  - add_cloud_metadata: ~
  - add_docker_metadata: ~
  - add_cluster_id: ~

```

**Review by oren-zohar — CHANGES_REQUESTED** [review]
Looks good. Shouldn't the deployment yaml go under `/deploy/kustomize/overlays/cloudbeat-aws`?

File: deploy/aws/cloudbeat-aws.yml (hunk 3 above)
Comment (uri-weisman) [reply deploy/aws/cloudbeat-aws.yml:20]: CSPM deployment won't be k8s related, that's why I thought it shouldn't reside under the `kustomize/overlays/` dir.

**Comment by uri-weisman** [comment]
>Shouldn't the deployment yaml go under `/deploy/kustomize/overlays/cloudbeat-aws`?

CSPM deployment won't be k8s related, that's why I thought it shouldn't reside under the `kustomize/overlays/` dir.

File: deploy/aws/cloudbeat-aws.yml (hunk 6 above)
Comment (uri-weisman) [reply deploy/aws/cloudbeat-aws.yml:55]: The `add_cluster_id` processor is not relevant for a non-k8s deployment, it will panic.

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**